fakenet/fakenet.py

```diff
@@ -284,17 +287,62 @@ def main():
                       action="store_true", dest="verbose", default=False,
                       help="print more verbose messages.")
     parser.add_option("-l", "--log-file", action="store", dest="log_file")
+    parser.add_option("-s", "--log-syslog", action="store_true", dest="syslog",
+                      default=False, help="Log to syslog via /dev/log")
     parser.add_option("-f", "--stop-flag", action="store", dest="stop_flag",
                       help="terminate if stop flag file is created")
+    # TODO: Rework the way loggers are created and configured by subcomponents
+    # to produce the expected result when logging control is asserted at the
+    # top level. For now, the setting serves its real purpose which is to ease
+    # testing on Linux after modifying logging such that console and file
+    # output are not mutually exclusive.
+    parser.add_option("-n", "--no-console-output", action="store_true",
+                      dest="no_con_out", default=False,
+                      help="Suppress console output (for testing on Linux)")
 
     (options, args) = parser.parse_args()
 
     logging_level = logging.DEBUG if options.verbose else logging.INFO
 
+    date_format = '%m/%d/%y %I:%M:%S %p'
+    logging.basicConfig(format='%(asctime)s [%(name)18s] %(message)s',
+                        datefmt=date_format, level=logging_level)
+    logger = logging.getLogger('')  # Get the root logger i.e. ''
+
+    if options.no_con_out:
+        logger.handlers = []
+
     if options.log_file:
-        logging.basicConfig(format='%(asctime)s [%(name)18s] %(message)s', datefmt='%m/%d/%y %I:%M:%S %p', level=logging.INFO, filename=options.log_file)
-    else:
-        logging.basicConfig(format='%(asctime)s [%(name)18s] %(message)s', datefmt='%m/%d/%y %I:%M:%S %p', level=logging.INFO)
+        try:
+            loghandler = logging.StreamHandler(stream=open(options.log_file,
+                                                           'a'))
+        except IOError:
+            print('Failed to open specified log file: %s' % (options.log_file))
+            sys.exit(1)
+        loghandler.formatter = logging.Formatter(
+            '%(asctime)s [%(name)18s] %(message)s', datefmt=date_format)
+        logger.addHandler(loghandler)
+
+    if options.syslog:
+        platform_name = platform.system()
+        sysloghandler = None
+        if platform_name == 'Windows':
+            sysloghandler = logging.handlers.NTEventLogHandler('FakeNet-NG')
```

Comment:
Running with `-s` on Windows without `pywin32` has no observable effect other than causing FakeNet-NG to emit the message `The Python Win32 extensions for NT (service, event logging) appear not to be available.`.

Reply:
Not including `pywin32` because it created some trouble last I looked into it and is a distraction from the scope and goals of this issue. The code to take advantage of the feature on Windows is an opportunistic inclusion as far as I am concerned. The effort of accommodating its requirements and testing its effectiveness is deferred until we've perfected everything else about FakeNet-NG and really yearn to see the Windows version log its output to the Windows event log. That said, when I did test this code with `pywin32`, it did work.